### ISSUE TITLE
[Exogenous tansformation] Lagged exogenous - target features

### ISSUE THREAD

**Issue by ncooder** [Nixtla/mlforecast#413]
### Description

Is there a way to create a feature from the exogenous and the target value with a lag and without a data leakage? Also in the AutoML can one add a list of exogenous transformations that can be passed for optuna to optimize which one gives better accuracy (also the original exogenous count)? 

### Use case

This would be very appreciated in AutoML models where the exogenous transformation is not simple to add.

**Comment by ncooder** [comment]
@jmoralez Assume I have an exogenous feature, I use transform_exog and lag it (I can also keep it as it is since I have future values). Then I will merge them with my target before passing it to the fit method. I want to use transformed target (lag it by 1) and create a new feature where I will use the exogenous and that transformed target and have like a ratio or a difference between them. Then after the fit I want to also have the values for each step, but for that I need to use before_predict_callback and extract the lagged target and again create the ratio/difference. The point is that is not possible to have this new feature in the recursive approach. I hope this explains the problem.

**Comment by jmoralez** [comment]
I see. If you want to create that feature during cross validation I think the easiest way is with a scikit-learn transformer that creates it in its transform method, that way it'll create it for the training set with the observed values and at each predict step.
